### ISSUE TITLE
Need for a proper architecture.

### ISSUE THREAD

**Issue by udhay24** [opendatakit/skunkworks-crow#225]
#### Software and hardware versions 
general issue regarding the codebase architecture.
#### Problem description
In skunkworks-crow we aren't using any specific architecture. but for the app to become production ready we need to have unit test codes which can be written only if we follow a proper architecture. i would like to propose a MVVM architecture since 

1. The ViewModel component is provided by google as a part of jetpack architecture components so i will be stable and future support will be provided 
2. ViewModel has life-cycle awareness. hence the boiler plate code can be reduced compared to MVP architecture.
3. it an be mocked hence writing unit test will be easier
4. it can be used with dagger 2 
#### Steps to reproduce the problem
not a problem
#### Expected behavior
discussion is needed before the code structure can be finalised

**Comment by shobhitagarwal1612** [comment]
Since we already have rxJava added to the project, it'd be nice to use that with MVVM instead of going forward with data-binding

**Comment by huangyz0918** [comment]
Refactoring always has a high cost, both in work and testing. This project doesn't have many network related stuff, and the business logic is also relatively less complex. I mean, refactor cannot bring much benefit, If you really want to do this job @udhay24 , I suggest starting it from some simple features, and merging after careful testing.

**Comment by udhay24** [comment]
> Refactoring always has a high cost, both in work and testing. This project doesn't have many network related stuff, and the business logic is also relatively less complex. I mean, refactor cannot bring much benefit, If you really want to do this job @udhay24 , I suggest starting it from some simple features, and merging after careful testing.

yeah even i was thinking the same and also it is difficult or sometimes not possible to write unit test cases for certain cases(for eg: in file transfer). but if we are only considering modularising the code by moving the logic to a viewmodel(for example)  the process can be easily achieved without much work and also will help in maintaining the code base.

**Comment by shobhitagarwal1612** [comment]
I agree with @huangyz0918. So the safest way to approach it would be do it many small portions rather than doing the entire refactor at once
